client/src/components/hero/Hero.tsx

```diff
@@ -17,64 +20,178 @@ interface HomeProps {
  * Hero for main home page.
  */
 const UnstyledHero: FunctionComponent<HomeProps> = ({ className }) => {
+  const { viewer } = useContext(ViewerContext);
+  // const announcementRef = useRef(0);
+
+  const [mainAnnouncementPosition, setMainAnnouncementPosition] = useState(0);
+  // created this for the case when the sub box needs to show the first announcement and the main box shows last announcement within the list of announcements
+  const [subAnnouncementPosition, setSubAnnouncementPosition] = useState(1);
+
+  const history = useHistory();
+
+  let announcementLength = 0;
+
+  const incrementAnnouncement = () => {
+    {
+      mainAnnouncementPosition >= announcementLength
+        ? setMainAnnouncementPosition(0)
+        : setMainAnnouncementPosition(mainAnnouncementPosition + 1);
+    }
+
+    {
+      subAnnouncementPosition >= announcementLength
+        ? setSubAnnouncementPosition(0)
+        : setSubAnnouncementPosition(subAnnouncementPosition + 1);
+    }
+  };
+
+  const decrementAnnouncement = () => {
+    {
+      mainAnnouncementPosition === 0
+        ? setMainAnnouncementPosition(announcementLength)
+        : setMainAnnouncementPosition(mainAnnouncementPosition - 1);
+    }
+    {
+      subAnnouncementPosition === 0
+        ? setSubAnnouncementPosition(announcementLength)
+        : setSubAnnouncementPosition(subAnnouncementPosition - 1);
+    }
+  };
+
+  if (!viewer?.announcements) {
+    return <div>loading...</div>;
+  } else {
+    announcementLength = viewer.announcements.length - 1;
```

Comment:
unnecssary declartion. use `useCallback` to memoize the function that I described above that is dependent on viewer and other necessary states and set the variable within that function `announcementLength = viewer.announcements.length - 1;`